lib/colors.ts

```diff
@@ -160,6 +162,43 @@ export function toPercent(value: number): string {
   return `${clean(100 * value)}%`
 }
 
+function isHex(sequence: number): RegExp {
+  return new RegExp(`^#?([\\da-f]{${sequence}})$`, 'i')
+}
+function isHex3(value: string): boolean {
+  return isHex(3).test(value)
+}
+function isHex4(value: string): boolean {
+  return isHex(4).test(value)
+}
+function isHex6(value: string): boolean {
+  return isHex(6).test(value)
+}
+export function isHex8(value: string): boolean {
+  return isHex(8).test(value)
+}
+
+function isHexNotation(value: string): boolean {
+  return [isHex3, isHex4, isHex6, isHex8].some(fn => fn(value))
```

Comment:
20 LOC is too much for small utility like this. Let’s go back to a simple RegExp.

Also, this method is not very effective since you are compiling the same RegExp over and over again.